static/app/actionCreators/pageFilters.tsx

```diff
@@ -293,9 +321,14 @@ export function initializeUrlState({
       pageFilters.datetime = getDatetimeFromState(storedState);
       shouldUsePinnedDatetime = true;
     }
+
+    if (!hasRepositoryInUrl && pinnedFilters.has('repository')) {
+      codecovPageFilters.repository = storedState.repository ?? null;
+    }
   }
 
   const {projects, environments: environment, datetime} = pageFilters;
+  const {repository} = codecovPageFilters;
```

Comment:
the indirection here is somewhat confusing

Reply:
I anticipate we'll be using 4 properties total with the next changes that add 3 more filters (codecov org, repo, branch, codecov date), so I went this way, but I can change it for now and adjust it later if you think it's useful